src/tsolvers/liasolver/LIACutSolver.cpp

```diff
@@ -0,0 +1,505 @@
+//
+// Created by prova on 28.08.18.
+//
+
+#include "LIACutSolver.h"
+#include "Matrix.h"
+
+///COMPUTING EQUALITY BASIS
+
+
+
+void LIACutSolver::pivotOnSmallestConsistentInPoly(LVRef var, Simplex& eqBasisSimplex) {
+    assert(eqBasisSimplex.isBasic(var));
+    LVRef v_found = LVRef_Undef;
+    for (auto const &term : eqBasisSimplex.getRowPoly(var)) {
+        auto var_nonbasic = term.var;
+        assert(eqBasisSimplex.isNonBasic(var_nonbasic));
+        assert(var_nonbasic != var);
+        auto const &coeff = term.coeff;
+        const bool is_coeff_zero = (coeff == 0);
+        if (!is_coeff_zero && (model.Lb(var_nonbasic) < model.Ub(var_nonbasic))) {
+            if (v_found != LVRef_Undef) {
+                v_found = var_nonbasic;
+                eqBasisSimplex.pivot(var, var_nonbasic);
+            }
+        }
+    }
+}
+
+void LIACutSolver::initialize(Simplex& eqBasisSimplex, LRAModel& eqBasisModel) {
+    //for all vars be it basic or non-basic
+    const vec<LVRef> & allVars = eqBasisModel.getAllVars();
+    for (int i = 0; i < allVars.size(); i++) {
+        LVRef var = allVars[i];
+        if (eqBasisModel.read(var) > eqBasisModel.Lb(var)) {
+            eqBasisModel.clearLBound(var);
+        }
+        if (eqBasisModel.read(var) < eqBasisModel.Ub(var)) {
+            eqBasisModel.clearUBound(var);
+        }
+        if ((eqBasisModel.read(var).hasDelta())) {
+            eqBasisModel.clearUBound(var);
+            eqBasisModel.clearLBound(var);
+        }
+    }
+
+    for (auto var : eqBasisSimplex.getBasicVars()) {
+        //for basicvar in basic
+        if (eqBasisModel.Lb(var) == eqBasisModel.Ub(var))
+            pivotOnSmallestConsistentInPoly(var, eqBasisSimplex);
+    }
+
+    for (int i = 0; i < allVars.size(); i++) {
+        LVRef var = allVars[i];
+        if (eqBasisModel.Lb(var) >= eqBasisModel.Ub(var))
+            continue;
+
+        //for x_k in basic union nonbasic
+        if (!eqBasisModel.Lb(var).isMinusInf()){//this checks if lower bound of var is minus infinite
+            //l_k := l_k +sigma
+            eqBasisModel.tightenLBound(var);
```

Comment:
This method does not seem to exist in current version of LRAModel.
Is some code missing?

Reply:
I can't seem to be able to comment the start of this thread, but (this version of) LIACutSolver.{cpp,h} should not have been added to the branch.  I'm removing it for now altogether, since the development of the code can be done in separation.